### PR TITLE
Documentation fix for concatenate

### LINKED ISSUE #11186: [DOC] Incorrect doxygen for cudf::concatenate
**Describe the bug**
Documentation for `cudf::concatenate` states: 
"Columns of tables_to_concat are concatenated vertically to return a single [table_view]"

However the return type of the function is `std::unique_ptr<table>`

**Steps/Code to reproduce bug**
https://docs.rapids.ai/api/libcudf/nightly/group__copy__concatenate.html#ga172353e7673decae3d2ebad09e003832

**Expected behavior**
Correct description

**Environment overview (please complete the following information)**
n/a

**Environment details**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by dagardner-nv**
cudf::concatinate returns a table not a table_view as the docstrings suggest

fixes #11186

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: cpp/include/cudf/concatenate.hpp [hunk 1]
```diff
@@ -83,8 +83,8 @@ std::unique_ptr<column> concatenate(
  *
  * @param tables_to_concat host_span of table views to be concatenated into a single table
  * @param mr Device memory resource used to allocate the returned table's device memory
- * @return A single table having all the rows from the elements of `tables_to_concat` respectively
- * in the same order.
+ * @return std::unique_ptr<table> single table having all the rows from the elements of
```
Comment (davidwendt) [inline cpp/include/cudf/concatenate.hpp:86]: Do not include the return type in the `@return` tag:
https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/docs/DOCUMENTATION.md#return

**Review by davidwendt — CHANGES_REQUESTED** [review]

**Review by davidwendt — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ajschmidt8** [comment]
ok to test

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**